apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,436 @@
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
+
+    const profileInfo: BasicProfileInterface = useSelector(
+        (state: AppState) => state.authenticationInformation.profileInfo
+    );
+    const phoneNumberSchema: ProfileSchema = useSelector((state: AppState) => {
+        const phoneNumberSchemas: ProfileSchema = state.authenticationInformation.profileSchemas.find(
+            (profileSchema: ProfileSchema) => profileSchema.name === "phoneNumbers");
+
+        if (phoneNumberSchemas && phoneNumberSchemas.subAttributes) {
+            const mobileSchema: ProfileSchema = phoneNumberSchemas.subAttributes?.find(
+                (subAttribute: ProfileSchema) => subAttribute.name === "mobile");
+
+            if (mobileSchema) return mobileSchema;
+        }
+
+        return phoneNumberSchemas;
+    });
+    const activeForm: string = useSelector((state: AppState) => state.global.activeForm);
+
+    const [ mobile, setMobile ] = useState<string>(undefined);
+    const [ editedMobile, setEditedMobile ] = useState<string>(undefined);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    let mobileType: string;
+
+    useEffect(() => {
+        if (isEmpty(profileInfo)) {
+            dispatch(getProfileInformation());
+        }
+    }, []);
+
+    const handleUpdate = (mobile: string) => {
+        setIsSubmitting(true);
+        const data: {
+            Operations: {
+                op: string;
+                value: Record<string, unknown>;
+            }[];
+            schemas: string[];
+        } = {
+            Operations: [
+                {
+                    op: "replace",
+                    value: {}
+                }
+            ],
+            schemas: [ "urn:ietf:params:scim:api:messages:2.0:PatchOp" ]
+        };
+
+        data.Operations[0].value = {
+            phoneNumbers: mobileType || phoneNumberSchema
+                ? [
+                    {
+                        type: mobileType || phoneNumberSchema.name,
+                        value: mobile
+                    }
+                ]
+                : [ mobile ]
+        };
+
+        updateProfileInfo(data)
+            .then(() => {
+                onAlertFired({
+                    description: t(
+                        "myAccount:components.accountRecovery.SMSRecovery.notifications.updateMobile" +
+                        ".success.description"
+                    ),
+                    level: AlertLevels.SUCCESS,
+                    message: t(
+                        "myAccount:components.accountRecovery.SMSRecovery.notifications.updateMobile.success.message"
+                    )
+                });
+
+                dispatch(getProfileInformation());
+                dispatch(setActiveForm(null));
+            })
+            .catch((error: AxiosError) => {
+                if (error.response && error.response.data && error.response.data.detail) {
```

Comment:
Simplify with null check
```suggestion
                if (error?.response?.data?.detail) {
```

Reply:
Addressed with https://github.com/wso2/identity-apps/commit/f82c47681cfe094407e95e1696b645a3aa799ef5